lib/galaxy/webapps/galaxy/controllers/workflow.py

```diff
@@ -319,6 +319,13 @@ def share(self, trans, id, email="", use_panels=False):
     @web.expose
     @web.require_login("Share or export Galaxy workflows")
     def sharing(self, trans, id, **kwargs):
+        """ Redirect the sharing workflow """
+        return trans.response.send_redirect(url_for('/') + 'workflows/share_workflow?id=' + id)
+
+    @web.expose
+    @web.require_login("Share or export Galaxy workflows")
+    @web.json
+    def sharing_workflow(self, trans, id, **kwargs):
```

Comment:
I think, we should use an api endpoint and remove this controller endpoint if possible.